src/dos/drive_virtual.cpp

```diff
@@ -536,20 +517,21 @@ bool Virtual_Drive::FindNext(DOS_DTA &dta)
 		if (cmp)
 			return true;
 	}
-	while (search_file) {
-		if (pos == search_file->onpos &&
-		    ((attr & DOS_ATTR_DIRECTORY) || !search_file->isdir) &&
-		    WildFileCmp(search_file->name.c_str(), pattern)) {
-			dta.SetResult(search_file->name.c_str(),
-			              search_file->size,
-			              search_file->date,
-			              search_file->time,
-			              (int)(search_file->isdir ? DOS_ATTR_DIRECTORY
-			                                       : DOS_ATTR_ARCHIVE));
-			search_file = search_file->next;
-			return true;
-		}
+	search_file = find_file_by_predicate(
+	        search_file, [pos, attr, pattern](vfile_block_t cur_file) {
+		        return pos == cur_file->onpos &&
+		               ((attr & DOS_ATTR_DIRECTORY) || !cur_file->isdir) &&
+		               WildFileCmp(cur_file->name.c_str(), pattern);
+	        });
```

Comment:
The embedded lambda is even harder to read than the original code, because it's mixing prototypes with the actual arguments.

Maybe the lamda could be constructed on its own, as a variable?

Also suggest replacing `cur_file` with `vfile`.